### ISSUE TITLE
Watch mode doesn't log existing errors on subsequent runs

### ISSUE THREAD

**Issue by nihalgonsalves** [microsoft/pyright#7267]
**Describe the bug**

Once watch mode has encountered a type error, it won't print it on subsequent runs (e.g. if you hit save again, modify another file, etc).

If you look at the logs you see:

```
1 error, 0 warnings, 0 informations
Watching for file changes...

...

0 errors, 0 warnings, 0 informations
Watching for file changes...
```

There's no indication that there are still existing errors - it looks like things were fixed. To really know if the codebase type checks properly you have to restart pyright and wait the entire initial load period before seeing the errors again.

In comparison `tsc --watch` will output existing errors on every new log line until you fix them, so when you see "0 errors" you know for sure that the entire project type checks successfully.

**Code or Screenshots**

```python
foo: int = 'hello'
# save -> '1 error' ...
# save again without fixing problem -> '0 errors' ...
```

**VS Code extension or command-line**

```bash
$ pyright --version
pyright 1.1.350
```

**Comment by erictraut** [comment]
The current behavior (which has been the same for nearly four years) is consistent with how it was designed. However, I think there's a reasonable argument in favor of changing it to be consistent with what you're describing. I suspect that very few pyright users make use of watch mode because the in-editor language server experience is preferable to most people.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
